File: Syncopaste/Assets/Scripts/StarSpawner.js
#pragma strict

public var prefabs: UnityEngine.GameObject[];
public var delay: float = 2;
public var delayRange: Vector2 = Vector2(0.2, 2);
public var active: int = 1;
private var screenUnitWidth: float;

function Start () {
	screenUnitWidth = (Screen.width / PixelPerfectCamera.pixelsToUnits) / 2;
	ResetDelay();
	StartCoroutine(StarGenerator());
}

function StarGenerator (): IEnumerator {
	yield WaitForSeconds (delay);
	if (active) {
		var newTransform = transform;
		newTransform.position.x = Random.Range(-screenUnitWidth, screenUnitWidth);
		Instantiate(prefabs[Random.Range(0, prefabs.Length)], newTransform.position, Quaternion.identity);
		ResetDelay();
	}
	
	StartCoroutine(StarGenerator());
}

function ResetDelay () {
	 delay = Random.Range(delayRange.x, delayRange.y);
}

File: Syncopaste/Assets/Scripts/StarSpawner.cs
using UnityEngine;
using System.Collections;

public class StarSpawner : MonoBehaviour {

	public GameObject[] prefabs;
	public CollisionManager collisionDelegate;
	public float delay = 2f;
	public Vector2 delayRange = new Vector2(.2f, 2);
	public bool active = true;

	private float screenUnitWidth;

	// Use this for initialization
	void Start () {
		screenUnitWidth = (Screen.width / PixelPerfectCamera.pixelsToUnits) / 2;
		ResetDelay();
		StartCoroutine(StarGenerator());
	}

	IEnumerator StarGenerator () {
		yield return new WaitForSeconds (delay);
		if (active) {
			var newTransform = transform;
			var position = newTransform.position;
			position.x = Random.Range(-screenUnitWidth, screenUnitWidth);
			newTransform.position = position;

			GameObjectUtil.Instantiate(prefabs[Random.Range(0, prefabs.Length)], newTransform.position);

			ResetDelay();
		}
		
		StartCoroutine(StarGenerator());
	}

	void ResetDelay() {
		delay = Random.Range(delayRange.x, delayRange.y);
	}
}
